src/formatDate.js

```diff
@@ -50,7 +53,65 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const prefixYear = (yearBefore, newFormat) => {
+    const yearUpdateFormat
+      = newFormat.filter(partDate => partDate.includes('YY'))[0];
+
+    const previousCentury = '19';
+    const currentCentury = '20';
+
+    if (yearBefore.toString().length > yearUpdateFormat.length) {
+      return yearBefore.slice(-2);
+    }
+
+    if (yearBefore.length === yearUpdateFormat.length) {
+      return yearBefore;
+    }
+
+    if (yearUpdateFormat.length === 4 && yearBefore >= 30) {
+      return previousCentury + yearBefore;
+    }
+
+    if (yearUpdateFormat.length === 4 && yearBefore < 30) {
+      return currentCentury + yearBefore;
+    }
+
+    return yearBefore;
+  };
+
+  let newFormatDate = '';
+
+  for (let i = 0; i < toFormat.length; i++) {
+    const dateArr = date.split(fromFormat[3]);
+    const indexYY = fromFormat.indexOf('YY') !== -1
+      ? fromFormat.indexOf('YY')
+      : fromFormat.indexOf('YYYY');
+
+    const indexMM = fromFormat.indexOf('MM');
+    const indexDD = fromFormat.indexOf('DD');
+
+    const day = dateArr[indexDD];
+    const month = dateArr[indexMM];
+    const year = dateArr[indexYY];
+
+    const separator = toFormat[toFormat.length - 1];
+
+    if (toFormat[i] === 'DD') {
+      newFormatDate += day + separator;
+    }
+
+    if (toFormat[i] === 'MM') {
+      newFormatDate += month + separator;
+    }
+
+    if (toFormat[i] === 'YY' || toFormat[i] === 'YYYY') {
+      const yearUpdate = prefixYear(year, toFormat);
+
+      newFormatDate += yearUpdate + separator;
+    }
+  }
```

Comment:
In JS strings are immutable, so each time you `+=` them you create new instances - which is an antipattern.

There are several ways to fix this - but try to "mutate" strings

As an alternative - you can use an array and then make `Array.join` or you can go with template literals

Reply:
thanks, all comment very relevant